eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -70,6 +70,10 @@ extends:
           - pwsh: |
               $tspConfigPathPattern = '^specification\/([^\/]+\/)+tspconfig\.yaml$'
               $readmePathPattern = '^specification\/([^\/]+\/){2,}readme\.md$'
+              if ('$(Build.Reason)' -eq 'PullRequest') {
```

Comment:
This is not needed. $SpecRepoCommit is set to `$(Build.SourceVersion)` which is already the head commit of the spec PR in PR case.

Reply:
We no longer use $(Build.SourceVersion) for APIView

Reply:
Can you remove this if it's no longer used?

Reply:
Ok. You need to move this assignment to step of `Generate SDK` as this value cannot be changed before the `sparse checkout` step.

Reply:
Why can this not be changed before the sparse checkout step?

Reply:
We need to check out against the merge commit in PR scenario instead of the head commit in the PR branch.